### PR TITLE
fixed central frequency parsing

### PR DESCRIPTION
**Pull request by AntoniaR**
This pull request fixes Issue #571 meaning that the observing frequency and band width in the headers of WSClean fits images can now be correctly parsed by TraP

### CODE REVIEW

**Comment by AntoniaR** [comment]
The checks failed in Travis because the TraP pip install is not working - I suspect that this is a module version error as it installed fine on my virtual machine. This should be resolved when TraP is migrated to python3

File: tkp/accessors/fitsimage.py [hunk 1]
```diff
@@ -124,23 +124,34 @@ def parse_frequency(self):
         try:
             header = self.header
             if header['TELESCOP'] in ('LOFAR', 'AARTFAAC'):
-                freq_eff = header['RESTFRQ']
-                if 'RESTBW' in header:
-                    freq_bw = header['RESTBW']
+                if 'RESTFRQ' in header:
+                    freq_eff = header['RESTFRQ']
+                    if 'RESTBW' in header:
+                        freq_bw = header['RESTBW']
```
Comment (ajstewart) [inline tkp/accessors/fitsimage.py:130]: You could probably tidy this up a bit if you drop this telescope line and just check for `RESTFRQ` and `RESTBW` in the header (maybe the LOFAR/AARTFAAC have been standardised such that you no longer need these?)

You could then avoid the duplication of the code below (the CTYPE3 and CTYPE4 checks).

Essentially: 
```
if `RESTFRQ` in header:

elif <CTYPE3 check>

elif <CTYPE4 check>

else <RESTFREQ fallback>
```

File: tkp/accessors/fitsimage.py [hunk 2]
```diff
@@ -124,23 +124,34 @@ def parse_frequency(self):
...
+                else:
+                    if ('CTYPE3' in header) and (header['CTYPE3'] in ('FREQ', 'VOPT')):
+                        freq_eff = header['CRVAL3']
+                        freq_bw = header['CDELT3']
+                    elif ('CTYPE4' in header) and (header['CTYPE4'] in ('FREQ', 'VOPT')):
+                        freq_eff = header['CRVAL4']
+                        freq_bw = header['CDELT4']
+                    else:
+                        freq_eff = header['RESTFREQ']
+                        freq_bw = 0.0
```
Comment (ajstewart) [inline tkp/accessors/fitsimage.py:145]: If the bandwidth is not found here it is set to 0.0 (same below), where as above it is set to 1.0. 

Setting to 0.0 can lead to unexpected behaviour as (from memory I think this is right) it doesn't calculate the metrics even if the other measurements have the same frequency. The 1.0 default as above is safer.

File: tkp/accessors/fitsimage.py [hunk 3]
```diff
@@ -124,23 +124,34 @@ def parse_frequency(self):
...
-                    freq_bw = header['cdelt4']
+                if ('CTYPE3' in header) and (header['CTYPE3'] in ('FREQ', 'VOPT')):
+                    freq_eff = header['CRVAL3']
+                    freq_bw = header['CDELT3']
+                elif ('CTYPE4' in header) and (header['CTYPE4'] in ('FREQ', 'VOPT')):
+                    freq_eff = header['CRVAL4']
+                    freq_bw = header['CDELT4']
                 else:
-                    freq_eff = header['restfreq']
+                    freq_eff = header['RESTFREQ']
```
Comment (ajstewart) [inline tkp/accessors/fitsimage.py:154]: Out of interest was it a conscious choice for this too be `RESTFREQ` and the above LOFAR one to be `RESTFRQ`?

File: tkp/accessors/fitsimage.py [hunk 4]
```diff
@@ -124,23 +124,34 @@ def parse_frequency(self):
...
                     freq_bw = 0.0
```
Comment (ajstewart) [inline tkp/accessors/fitsimage.py:155]: Other instance of bw being set to 0.0.

File: tkp/accessors/fitsimage.py (hunk 1 above)
Comment (AntoniaR) [reply tkp/accessors/fitsimage.py:130]: Thanks for checking the code so quickly! 
That looks like an excellent idea. LOFAR and AARTFAAC12 are typically using WSClean now so should be standard, but lets keep RESTFRQ and RESTFREQ for old images. I'll make those changes today

File: tkp/accessors/fitsimage.py (hunk 2 above)
Comment (AntoniaR) [reply tkp/accessors/fitsimage.py:145]: I've found that 0 works, but basically it then sets up a new frequency band for every image. So agreed this is not desirable behaviour though there is also a work around in job_params.cfg. 1 also isn't great but avoids unexpected 0 problems... So, agreed, I will set these to 1 as well.

File: tkp/accessors/fitsimage.py (hunk 3 above)
Comment (AntoniaR) [reply tkp/accessors/fitsimage.py:154]: This is legacy code and really dependent on what you use to make the images. So best to keep for different imagers.

File: tkp/accessors/fitsimage.py (hunk 4 above)
Comment (AntoniaR) [reply tkp/accessors/fitsimage.py:155]: Agreed

**Comment by AntoniaR** [comment]
I have made the changes that Adam suggested and updated the branch Issue571.

**Comment by AntoniaR** [comment]
Great, thanks!

**Pull request merged**